### PR TITLE
Sanitize and render comments with rich text

### PR DESCRIPTION
**Pull request by tillyw**
## Associated issues
fixes https://github.com/cityofaustin/atd-data-tech/issues/9495

## Testing
**URL to test:** 
https://deploy-preview-689--atd-moped-main.netlify.app/

**Steps to test:**
1. Add status updates to an existing project (for both a regular project and a signal project)
2. Follow the project and add yourself to the team
3. Go to comments tab and add formatting and an emoji to the comment 
4. Check the following views to ensure rich text is displayed: project summary, comments tab, all projects view, signals projects view, dashboard view (my projects and following)

Note: Rich text is already sanitized before being saved in comments view. Added sanitization when adding a new status update on summary page.

---
#### Ship list
- [x] Code reviewed 
- [x] Product manager approved
- [x] Added to [QA test script](https://docs.google.com/spreadsheets/d/1n_O6MLh9cwwPf57HUM394Ea-z9uuoEV1-QW4axNZXLE/edit#) if applicable

Screen Shot 2022-06-21 at 5 18 07 PM
Screen Shot 2022-06-21 at 5 17 58 PM
Screen Shot 2022-06-21 at 5 17 49 PM
Screen Shot 2022-06-21 at 5 17 39 PM
Screen Shot 2022-06-21 at 5 17 30 PM
Screen Shot 2022-06-21 at 5 17 15 PM

### CODE REVIEW

**Review by amenity — APPROVED** [review]
Works beautifully, @tillyw! 🧼 ✍🏻

**Review by frankhereford — APPROVED** [review]
Thanks Tilly!

I am unable to get the app to consume and reproduce anything that is being evaluated in an unescaped context - this is working great. 🌟 

Seeing those regex's leave the codebase is comforting too - tokenizing is hard™ work! 🙏 

I observed the smallest unexpected behavior that you may want to look at or document in a backlog issue, but it's very niche **_to say the least_**. Here's how I am able to reproduce it:

* Navigate to the Project Summary tab
* Create a new status update using the circle-plus icon
* Enter the following characters: `<bold>not bold</bold>`
* Observe this drawn on screen:
image


* Navigate to the comments tab
* Observe the same comment drawn as:
image

* Navigate back to the summary screen
* And finally observe the same comment drawn as follows:
image


I think this demonstrates that there is something different in how that string is handled when it's provided from the user and when it's provided from the database. 

I think there is some sort of complexity in using html-react-parse as sanitizer or special-character escaping mechanism, but I am still comfortable in the system's ability to prevent malicious attacks. 🚢 🚢

**Comment by tillyw** [comment]
> Thanks Tilly!
> 
> I am unable to get the app to consume and reproduce anything that is being evaluated in an unescaped context - this is working great. 🌟
> 
> Seeing those regex's leave the codebase is comforting too - tokenizing is hard™ work! 🙏
> 
> I observed the smallest unexpected behavior that you may want to look at or document in a backlog issue, but it's very niche **_to say the least_**. Here's how I am able to reproduce it:
> 
> * Navigate to the Project Summary tab
> * Create a new status update using the circle-plus icon
> * Enter the following characters: `<bold>not bold</bold>`
> * Observe this drawn on screen:
> 
> image
> 
> * Navigate to the comments tab
> * Observe the same comment drawn as:
> 
> image
> 
> * Navigate back to the summary screen
> * And finally observe the same comment drawn as follows:
> 
> image
> 
> I think this demonstrates that there is something different in how that string is handled when it's provided from the user and when it's provided from the database.
> 
> I think there is some sort of complexity in using html-react-parse as sanitizer or special-character escaping mechanism, but I am still comfortable in the system's ability to prevent malicious attacks. 🚢 🚢

Thanks for bringing this to my attention @frankhereford! After some troubleshooting I realized the issue is due to the fact that I was purifying the user input as the user was typing it. I corrected this by purifying it on save instead. Can you please take another look and let me know if that addresses the issue while ensuring the user input is safe? Thanks again!

File: moped-editor/src/views/dashboard/DashboardView.js [hunk 1]
```diff
@@ -139,13 +139,11 @@ const DashboardView = () => {
       project["current_status"] = project.project.current_status;
 
       // project status update equivalent to most recent project note
+      // html is parsed before being rendered in the DashboardEditModal component
```
Comment (chiaberry) [inline moped-editor/src/views/dashboard/DashboardView.js:142]: Where is it being parsed?

Comment (tillyw) [reply moped-editor/src/views/dashboard/DashboardView.js:142]: it is sent into the DashboardEditModal component [here](https://github.com/cityofaustin/atd-moped/blob/ae264a5d4ba55e8639d0fd48c4f92564f6ac42db/moped-editor/src/views/dashboard/DashboardView.js#L211) and parsed [here](https://github.com/cityofaustin/atd-moped/blob/ae264a5d4ba55e8639d0fd48c4f92564f6ac42db/moped-editor/src/views/dashboard/DashboardEditModal.js#L48)

File: moped-editor/src/views/projects/projectView/ProjectSummary/ProjectSummaryStatusUpdate.js [hunk 2]
```diff
@@ -93,6 +95,7 @@ const ProjectSummaryStatusUpdate = ({ projectId, data, refetch, classes }) => {
    * Handles saving the status update
    */
   const handleStatusUpdateSave = () => {
+    console.log("save new status")
```
[the remark this answers is not in the document]
Comment (tillyw) [reply moped-editor/src/views/projects/projectView/ProjectSummary/ProjectSummaryStatusUpdate.js]: done, thanks!

**Review by mateoclarke — COMMENTED** [review]
👍

**Pull request merged**